src/main/java/be/syntra/devshop/DevshopFront/services/ProductService.java

```diff
@@ -26,6 +26,8 @@
 
     Product findById(Long id);
 
+    ProductList findAllWithCorrespondingCategory(Long id);
```

Comment:
findAllByCategory() ?

Reply:
changed to findAllWithOnlyCategory, because it'll find all products which ONLY have the corresponding categoryID on them